ios/FluentUI/Vnext/Core/Theming.swift

```diff
@@ -51,23 +51,80 @@ class MSFTokensBase {
     private var _theme: FluentUIStyle?
 }
 
-// MARK: Theme SwiftUI Environment Value
+// MARK: SwiftUI Environment Values
 
 struct ThemeKey: EnvironmentKey {
     static var defaultValue: FluentUIStyle {
         return FluentUIThemeManager.S
     }
 }
 
+struct WindowProviderKey: EnvironmentKey {
+    static var defaultValue: FluentUIWindowProvider? {
+        return nil
+    }
+}
+
 extension EnvironmentValues {
     var theme: FluentUIStyle {
         get { self[ThemeKey.self] }
         set { self[ThemeKey.self] = newValue }
     }
+
+    var windowProvider: FluentUIWindowProvider? {
+        get { self[WindowProviderKey.self] }
+        set { self[WindowProviderKey.self] = newValue }
+    }
 }
 
 extension View {
-    func usingTheme(_ theme: FluentUIStyle) -> some View {
+
+    /// Overrides a theme for a specific SwiftUI View and its view hierarchy.
+    /// - Parameter theme: Instance of the custom overriding theme.
+    /// - Returns: The view with its theme environment value overriden.
+    func customTheme(_ theme: FluentUIStyle) -> some View {
         environment(\.theme, theme)
     }
+
+    /// Sets the window provider used to compute the theme associated with the SwiftUI View
+    /// based on the window that it belongs to.
+    /// - Parameter windowProvider: An instance of an class that implements the FluentUIWindowProvider protocol.
+    /// - Returns: The view with the window provider set for itself and its view hierarchy.
+    func windowProvider(_ windowProvider: FluentUIWindowProvider?) -> some View {
+        environment(\.windowProvider, windowProvider)
+    }
+
+    /// SwiftUI Views using this modifier need to define the following environment values:
+    ///  - @Environment(\.theme) var theme: FluentUIStyle
+    ///  - @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    ///
+    /// The instances of these environment value properties will be passed to the SwiftUI view's subviews in its hierarchy.
+    ///
+    /// The logic behind the order or precedence to retrieve the correct theme to be used is:
+    ///  1. If the SwiftUI View gets a non-default theme instance through the environment value property,
```

Comment:
If a `SwiftUI` container has an Objective-C subview, How will the subview be notified about the theme changes? I am assuming this precedence is applied for purely native SwiftUI components.

Reply:
The SwiftUI View that is exposed by the UIView will get the theme change through the token subscription to the theme change event. And yes, the theme override for a specific View and its view hierarchy is something supported only on the SwiftUI hierarchy.